solr/core/src/java/org/apache/solr/search/SolrCache.java

```diff
@@ -106,8 +107,23 @@
    * @return existing or newly computed value, null if there was no existing value and
    * it was not possible to compute a new value (in which case the new mapping won't be created).
    */
+  // Can we deprecate/remove this in favor of the other signature?
   public V computeIfAbsent(K key, Function<? super K, ? extends V> mappingFunction);
 
+  default V computeIfAbsent(K key, IOFunction<? super K, ? extends V> mappingFunction) throws IOException {
+    try {
+      return computeIfAbsent(key, (Function<? super K, ? extends V>) k -> {
+        try {
+          return mappingFunction.apply(k);
+        } catch (IOException e) {
+          throw new WrappingException(e);
```

Comment:
Can also pass this along via an AtomicReference or possible via Lucene's Rethrows class if people would prefer to see that.